openupgradelib/openupgrade.py

```diff
@@ -2258,3 +2259,16 @@ def chunked(records, single=True):
                 yield record
             continue
         yield chunk
+
+
+def switch_noupdate(env, xml_ids, value):
+    for xml_id in xml_ids:
+        if '.' not in xml_id:
+            logger.error(
+                'Need the module reference to be specified in the IDs' % xml_id)
+        else:
+            env.cr.execute("""
+                UPDATE ir_model_data
+                SET noupdate = NOT COALESCE(noupdate, FALSE)
```

Comment:
Must be `value`

Reply:
Then the name of the method is wrong. "switch" means change the noupdate **always**. If you want a method that only puts the passed value, then the name "switch" is wrong, as sometimes you only want to assure a value but you really are not changing it 😏